test/test_indexable_types.cpp

```diff
@@ -0,0 +1,96 @@
+//------------------------------------------------------------------------------
+// © 2025. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#include <array>
+
+#include <ports-of-call/portability.hpp>
+#include <singularity-eos/base/indexable_types.hpp>
+
+#ifndef CATCH_CONFIG_FAST_COMPILE
+#define CATCH_CONFIG_FAST_COMPILE
+#include <catch2/catch_test_macros.hpp>
+#endif
+
+using namespace singularity::IndexerUtils;
+using namespace singularity::IndexableTypes;
+
+using Lambda_t = VariadicIndexer<MeanIonizationState, ElectronFraction, LogDensity>;
+
+class ManualLambda_t {
+ public:
+  ManualLambda_t() = default;
+  Real &operator[](const std::size_t idx) { return data_[idx]; }
```

Comment:
Will things work if a lambda is defined without `Real &operator[](const std::size_t idx);`?

Reply:
Yes---so long as it contains all types needed.